lib/confusables/generate_list.go

```diff
@@ -47,24 +48,64 @@ func isAllowed(from, to string) bool {
 		}
 	}
 	for _, rn := range toRune {
-		if unicode.In(rn, BasicLatin) || unicode.IsSpace(rn) {
-			return true
+		if !unicode.In(rn, BasicLatin) {
+			return false
 		}
 	}
 
-	return false
+	return true
 }
 
 func formatUnicodeIDs(ids string) string {
 	var formattedIDs string
 	for _, charID := range strings.Split(ids, " ") {
-		newID := fmt.Sprintf("\\U%s%s", strings.Repeat("0", 8-len(charID)), charID)
-		formattedIDs += newID
+		// newID := fmt.Sprintf("\\U%s%s", strings.Repeat("0", 8-len(charID)), charID)
+		i, err := strconv.ParseInt(charID, 16, 32)
+		if err != nil {
+			fmt.Println(err)
+			continue
+		}
+
+		c := rune(int32(i))
+
+		formattedIDs += string(c)
 	}
 
 	return formattedIDs
 }
 
+func regexReplace(regex, replace, replaced string) string {
+	newRegex := regexp.MustCompile(regex)
+	reReplaced := newRegex.ReplaceAllString(replace, replaced)
+
+	if reReplaced != replace {
+		return reReplaced
+	}
+
+	return replace
+}
```

Comment:
Consider inlining this function -- it doesn't make much sense having it be standalone when it's essentially a two-liner IMO.

Reply:
im dumb, wdym by inlining it

Reply:
Remove the function and write `r := regexp.MustCompile(regex)`, `r.ReplaceAllString(..., ...)` directly at the callsite instead.

Reply:
Ah, well the regex won't even be used anymore because it's being replaced with a string replacer as you suggested